### PR TITLE
script to modify the state history retention period via the safe sdk

### LINKED ISSUE #2008: modify the `stateHistoryRetentionPeriod` in via the safe sdk
only the admin , a safe multisig wallet, can modify the `stateHistoryRetentionPeriod`. 
use the safe sdk to create the script that can be ran when Espresso wants to modify the `stateHistoryRetentionPeriod`.

### PR DESCRIPTION
**Pull request by alysiahuggins**
Closes #2008

### This PR:
- adds a script to modify the state history retention period via the safe sdk so that the admin, which is a multisig wallet, can execute the transaction
- adds instructions in the readme above

### CODE REVIEW

File: contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    await proposeSetStateHistoryRetentionTransaction(
+      safeSdk,
+      safeService,
+      orchestratorSignerAddress,
+      safeAddress,
+      stateHistoryRetentionPeriod,
+    );
+
+    console.log(
+      `The other owners of the Safe Multisig wallet need to sign the transaction via the Safe UI https://app.safe.global/transactions/queue?safe=sep:${safeAddress}`,
```
Comment (Sneh1999) [inline contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts:45]: nit: `Proposal created successfully. The other owners of the Safe Multisig wallet need to sign the transaction via the Safe UI https://app.safe.global/transactions/queue?safe=sep:${safeAddress}`

File: contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    senderSignature: senderSignature.data,
+  });
+}
+
+/**
+ * Function to create the transaction data for setting the stateHistoryRetentionPeriod
+ * @param {number} retention_period - The state history retention period in seconds
+ * @returns {string} - Encoded transaction data
+ */
+function createStateHistoryRetentionPeriodTxData(retention_period: number): string {
```
Comment (Sneh1999) [inline contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts:97]: nit: does this have to be a function? it is just two lines so we can prob merge this with `proposeSetStateHistoryRetentionTransaction`

File: contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+ */
+async function createSafeTransaction(
+  safeSDK: Safe,
+  contractAddress: string,
+  data: string,
+  value: string,
+): Promise<LocalSafeTransaction> {
+  // Prepare the safe transaction data with the contract address, data, and value
+  let safeTransactionData = createSafeTransactionData(contractAddress, data, value);
+  console.log("data hex: ", data);
```
Comment (Sneh1999) [inline contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts:123]: `console.log("Safe Transaction Data hex: ", data)`

File: contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+/**
+ * Creates a Safe transaction object
+ *
+ * @param {Safe} safeSDK - An instance of the Safe SDK
+ * @param {string} contractAddress - The address of the contract to interact with
+ * @param {string} data - The data payload for the transaction
+ * @param {string} value - The value to be sent with the transaction
+ * @returns {Promise<any>} - A promise that resolves to the Safe transaction object
+ */
+async function createSafeTransaction(
```
[the remark this answers is not in the document]
Comment (alysiahuggins) [reply contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts]: thanks, addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/a2ab0f54a13689832cadbd0500193e1cf3ebeb75#diff-ed9be146bb82143e9fdc2e1e482dde2e0977637db45081c1b52d42ad64ca578fR66

File: contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts (hunk 1 above)
Comment (alysiahuggins) [reply contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts:42]: thanks, addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/a2ab0f54a13689832cadbd0500193e1cf3ebeb75#diff-ed9be146bb82143e9fdc2e1e482dde2e0977637db45081c1b52d42ad64ca578fR66

File: contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts (hunk 3 above)
Comment (alysiahuggins) [reply contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts]: thanks, addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/78da4361f5626bb7cf5e9e82e04b4c0fe5f8419e

**Review by Sneh1999 — COMMENTED** [review]
LGTM!

**Review by Sneh1999 — APPROVED** [review]

**Pull request merged**